ChemistryLib/CreatePhreeqcInterface.cpp

```diff
@@ -0,0 +1,84 @@
+#include "PhreeqcInterfaceData/KineticReactant.h"
+#include "PhreeqcInterfaceData/ReactionRate.h"
+#include "PhreeqcInterfaceData/Solution.h"
+
+namespace ChemistryLib
+{
+std::unique_ptr<phreeqcIO> createPhreeqcInterface(
+    std::size_t const num_nodes,
+    std::vector<std::pair<int, std::string>> const& transport_process_id_map,
+    boost::optional<BaseLib::ConfigTree> const& config)
+{
+    // database
+    //! \ogs_file_param{prj__chemical_system__database}
+    auto const database = config->getConfigParameter<std::string>("database");
+
+    // solution
+    //! \ogs_file_param{prj__chemical_system__solution}
+    auto const solution = parseSolution(config->getConfigSubtree("solution"));
+
+    auto const& components = solution.components;
+    for (auto const& component : components)
+    {
+        auto transport_process_id_pair = std::find_if(
+            transport_process_id_map.begin(), transport_process_id_map.end(),
+            [&component](std::pair<int, std::string> const& id_pair) {
+                return id_pair.second == component.name;
+            });
+
+        if (transport_process_id_pair == transport_process_id_map.end())
+            ERR("Component %s given in <solution>/<components> is not found in "
+                "specified coupled processes (see "
+                "<process>/<process_variables>/<concentration>).",
+                component.name);
+    }
+    if (components.size() + 1 != transport_process_id_map.size())
+        ERR("The number of components given in <solution>/<components> is not "
+            "in line with the number of transport processes - 1 which stands "
+            "for the transport process of hydrogen.");
+
+    std::vector<Solution> solutions(num_nodes, solution);
+
+    // equilibrium phases
+    //! \ogs_file_param{prj__chemical_system__equilibrium_phases}
+    auto const nodal_equilibrium_phases = parseEquilibriumPhases(
+        config->getConfigSubtreeOptional("equilibrium_phases"));
+    std::vector<std::vector<EquilibriumPhase>> equilibrium_phases(
+        num_nodes, nodal_equilibrium_phases);
+
+    // kinetic reactants
+    //! \ogs_file_param{prj__chemical_system__kinetic_reactants}
+    auto const nodal_kinetic_reactants = parseKineticReactants(
+        config->getConfigSubtreeOptional("kinetic_reactants"));
+    std::vector<std::vector<KineticReactant>> kinetic_reactants(
+        num_nodes, nodal_kinetic_reactants);
+
+    // rates
+    //! \ogs_file_param{prj__chemical_system__rates}
+    auto const reaction_rates =
+        parseReactionRates(config->getConfigSubtreeOptional("rates"));
+
+    // output
+    auto output = createOutput(components, nodal_equilibrium_phases,
+                               nodal_kinetic_reactants);
+
+    return std::make_unique<phreeqcIO>(database, solutions, equilibrium_phases,
+                                       kinetic_reactants, reaction_rates,
+                                       output, transport_process_id_map);
```

Comment:
Several `std::move` would be beneficial here to avoid copies.
Please make yourself familiar with the move-semantics, especially what does it mean for a type to be "moved".

Reply:
Dropped a few unneeded utilization of `std::move`. Need some time to fully understand in which situation `std::move` shall be used or not.

Reply:
Still not changed.

Reply:
Fixed.

Reply:
`transport_process_id_map` is still copied twice. One copy is OK, two are not necessary.

Reply:
Fixed.